### PR TITLE
chore: separate out js functionality for each component

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

 Continue migrating the JS into enhancement.js files for the components they relate to

## Files Changed:

// Load Inputgroup functionality from components/Inputgroup/metadata/enhancement.js
// Load Stepper functionality from components/stepper/metadata/enhancement.js
// Load Picker functionality from components/picker/metadata/enhancement.js
// Load Treeview functionality from components/treeview/metadata/enhancement.js
// Load Accordion functionality from components/accordion/metadata/enhancement.js
// Load Cyclebutton functionality from components/cyclebutton/metadata/enhancement.js
// Load Sliders functionality from components/slider/metadata/enhancement.js
// Load Dial functionality from components/dial/metadata/enhancement.js
// Load Dialog functionality from components/dialog/metadata/enhancement.js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
   },
   "scripts": {
     "build": "yarn build:components",
-    "build:all": "run-s build:components build:site build:preview mv:preview",
+    "build:all": "run-s build:components build:preview mv:preview",
```
Comment (castastrophe) [inline package.json:25]: Can you pull this out from this PR before we merge? I'd love to keep it scoped to just the *.js addition

**Review by castastrophe — CHANGES_REQUESTED** [review]
Two small updates but otherwise looking awesome - should be able to merge this tomorrow.

**Review by pfulton — COMMENTED** [review]
I'm sure you realize this and just haven't gotten to take care of it yet, but when I run this locally and try to view the docs site, I'm getting a `setPickerValue is not a function` error in the console, and it's preventing anything else on any of the component pages from loading properly.

Again, you probably know about this already, but I thought I should call it out so that we can take care of it!

File: components/picker/metadata/enhancement.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      }
+    }
+  } else {
+    if (openPicker) {
+      toggleOpen(openPicker, false)
+    }
+  }
+})
+
+window.setPickerValue = setPickerValue
```
Comment (castastrophe) [inline components/picker/metadata/enhancement.js:187]: I suspect we can nix this now

**Comment by castastrophe** [comment]
> I'm sure you realize this and just haven't gotten to take care of it yet, but when I run this locally and try to view the docs site, I'm getting a `setPickerValue is not a function` error in the console, and it's preventing anything else on any of the component pages from loading properly.
> 
> Again, you probably know about this already, but I thought I should call it out so that we can take care of it!

Great catch Patrick! I suspect this comes from needing to load the enhancement.js assets first before `tools/documentation/assets/js/site.js` which references it. We'll also need a more graceful degradation if the function can't be found - an if check should do the trick to see if window.setPickerValue is undefined.

File: components/dial/metadata/enhancement.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    dial.addEventListener("mousedown", onMouseDown)
+  }
+}
+
+function enhanceAll() {
+  Array.prototype.forEach.call(
+    document.querySelectorAll(".spectrum-Dial"),
+    function (dial) {
+      makeDial(dial)
+    }
```
[the remark this answers is not in the document]
Comment (Rajdeepc) [reply components/dial/metadata/enhancement.js:36]: Done

File: package.json (hunk 1 above)
Comment (Rajdeepc) [reply package.json:25]: Sure Done.

**Pull request merged**